### PR TITLE
Add status update to Project List View 

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/9044

Adds status update to columns as well as filtering. 

## Testing
**URL to test:** 

test locally or moped test. 

This branch branches off of the funding source branch, so if you deploy this branch and test, you will see both enhancements on the deployment. 

**Steps to test:**

https://moped-test.austinmobility.io/moped/projects/174?tab=summary

Add a status to the summary page. See it on the project list view. 

Another test to do is to add an internal note on the comments page and check that it does not show up as a status. 

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     LEFT JOIN jsonb_array_elements(mp.task_order) as task_order_filter ON true
+     LEFT JOIN moped_proj_components mpc ON mpc.project_id = mp.project_id and mpc.status_id = 1
+     LEFT JOIN moped_proj_features mpf ON mpc.project_component_id = mpf.project_component_id
+     LEFT JOIN public.moped_project_types mpt ON mpt.project_id = mp.project_id
+     LEFT JOIN public.moped_types mt ON mpt.project_type_id = mt.type_id
+     LEFT JOIN moped_proj_notes mpn ON mpn.project_note_id = 
+      (SELECT mpn2.project_note_id FROM moped_proj_notes mpn2 
+        WHERE mpn2.project_id = mp.project_id and mpn2.project_note_type = 2
+        ORDER BY mpn2.date_created desc
+      LIMIT 1)
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql:76]: lines 72-76

**Review by frankhereford — CHANGES_REQUESTED** [review]
I have tested this feature locally and it works great.

I have a change-request below regarding the query that defines the view. I think it may be able to be optimized by moving the subselect into the select clause propper.

However, I think it should be noted that at our scale, the optimization is only going to pick us up a few milliseconds. Pedantically, the query is maybe a magnitude slower than it was previously, but still clocking in at ~25 ms which is wicked fast.

**Review by mateoclarke — APPROVED** [review]
🚢 💨 ✨ 

I defer to Frank's suggestions, and just want to reiterate that I'm grateful for both of y'alls knowledge and persistence to implement, profile, optimize, and iterate on this view query.

File: moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql]: I updated the query that defines the view. And after turning off JIT (https://github.com/cityofaustin/atd-moped/pull/648) it is going super quick. Thank you!

**Review by frankhereford — APPROVED** [review]

**Review by patrickm02L — CHANGES_REQUESTED** [review]
The status does not delete from the status column in the projects table once deleted in the other places it's been added. 

https://user-images.githubusercontent.com/5566727/168612051-bd64f177-6752-4941-99a1-745dea028d3f.mp4

File: moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      ELSE mp.current_status
+    END AS status_name,
+    string_agg(task_order_filter->>'display_name', ',') as task_order_name,
+    mp.contractor,
+    mp.purchase_order_number,
+    json_agg(mpf.feature) as project_feature,
+    mt.type_name,
+    fsl.funding_source_name,
+    (SELECT mpn.project_note FROM moped_proj_notes mpn
+      WHERE mpn.project_id = mp.project_id and mpn.project_note_type = 2 and mpn.status_id = 1
```
Comment (chiaberry) [inline moped-database/migrations/1651513363000_alter_project_list_view_status/up.sql:62]: updated this line to also check that status has not been soft deleted.

**Comment by chiaberry** [comment]
@patrickm02L similar issue as the previous PR, i wasn't checking for soft delete status. Updated the code to only get active status updates.

**Comment by mateoclarke** [comment]
Added "Add a status update on the Summary tab. Verify it renders correctly in the Projects table" to the QA Script doc

**Pull request merged**